base/show.jl

```diff
@@ -1542,9 +1542,18 @@ end
 showcompact(x) = showcompact(STDOUT, x)
 function showcompact(io::IO, x)
     if get(io, :compact, false)
```

Comment:
If it's generally considered a good idea to check whether a flag is set to avoid setting it twice without reason, shouldn't we do this automatically under the hood from `IOContext()`? That would make this code much shorter.